### ISSUE TITLE
bytesIn/bytesOut

### ISSUE THREAD

**Issue by AVitg** [kamushadenes/cefevent#17]
Hey @kamushadenes - I just stumbled across this valuable project, really appreciate your work.
I recently figured out, that in the current CEF definition, bytesIn and bytesOut is allowed to be Long from CEF version 1.0 onwards
(see V26 of CEF implementation guide)

do you have any thoughts of implementing these two different standards?
Cheers
A

**Comment by kamushadenes** [comment]
Hello @AVitg, thank you for the kind words.

It has been a long time since I used CEF, but I implemented the changes you requested. Version 0.5.5 is already on PyPi.

Please let me know if it's working accordingly.

**Comment by AVitg** [comment]
Thanks for the very quick fix, saw you added several lines of code, appreciate it.
so yes, works for me, and I learned something new: "int" in Python 3 is equivalent to "long" in Python 2
see https://docs.python.org/3/whatsnew/3.0.html#integers

did not test your code on python 2 - if it runs, the below would return not the expected "long" value, but this is only a minor for me ;)

`    def _validate_field_value(self, field: AnyStr, value: Any):
        obj = self._reverse_extension_dictionary[field]

        for dt in obj["data_type"]:
            if dt in ["Integer", "Long"]:
                try:
                    return int(value)
`

**Comment by kamushadenes** [comment]
It indeed targets Python 3. It's possible to make it work in both, but Python 2 was [officially killed in January 2020](https://www.python.org/doc/sunset-python-2/). 

Glad it worked for you, and feel free to open another issue if you need.

**Comment by AVitg** [comment]
whilst thinking about this... if the data_type is defined as "int" ... and somebody would use a "long" number (not in bytesIn/Out, but somewhere else...)  it would work in python, but create issues in the ArcSight world... (sorry i digged too deep ;) )

**Comment by kamushadenes** [comment]
Very well noted, I added a check for that!

Both in Python 2 and in ArcSight, ints are defined as 2^31-1, so I did this:

```python
            if dt in ["Integer", "Long"]:
                if dt == "Integer" and value > 2**31-1:
                    continue
                try:
                    return int(value)
                except:
                    continue
```

I also added an extra check for the `*Port` fields (like `sourcePort`), ensuring they are between 0 and 65535.

Those fixes (and your PR, thanks for it!) are in 0.5.6.

**Comment by AVitg** [comment]
Thanks for your time on the project!
